subsys/bluetooth/host/settings.c

```diff
@@ -19,7 +19,7 @@
 #include "hci_core.h"
 #include "settings.h"
 
-#if defined(BT_SETTINGS_USE_PRINTK)
+#ifdef CONFIG_BT_SETTINGS_USE_PRINTK
```

Comment:
Good fix, but could you please keep the `#if defined(...)` style since that's the convention throughout the Bluetooth code base?

Reply:
Alright, will change back.

Not sure how that `#if defined(...)` thing took off though. `#ifdef` is the oldest meme in all of C. :)

Reply:
`#if defined(..)` is a bit more flexible since you can chain multiple conditions using `#if defined(FOO) && defined(BAR)` etc. If we'd go with Linux style I think the preferred would actually be `#if IS_ENABLED(CONFIG_...)`.

Reply:
Changed

Reply:
A regular `if (IS_ENABLED(...))` might be the nicest when possible. Avoids bitrot in the body, since it the compiler sees it, and it still gets optimized away (though there might be a dummy empty function for stuff that crosses compilation units unless there's link-time optimization).

Saw there were some changes like that to the bluetooth code earlier. :ok_hand:

Don't think `IS_ENABLED()` is used much outside functions.

Reply:
Absolutely agree, but I *have* seen it used fairly extensively for pre-processor code in Linux as well

Reply:
Btw, @rakons pointed out a subtle difference between `defined` and `IS_ENABLED` here: https://github.com/zephyrproject-rtos/zephyr/pull/19407#issuecomment-541698063

Reply:
Usually won't matter for Kconfig symbols at least, because `n`-valued symbols don't generate any output in the generated header.

There's a similar subtle thing with `#if FOO` vs. `#ifdef FOO` too. For `#define FOO 0`, `#if FOO` is false and `#ifdef FOO` is true.

Undefined symbols evaluate to 0 too, so `#if SOME_UNDEFINED_SYMBOL == 0` is true. Can lead to gotchas when testing `int` symbols and stuff...